.github/workflows/ci.yml

```diff
@@ -40,3 +39,37 @@ jobs:
 
       - name: Run tests
         run: cargo test
+
+  lint:
+    name: Run linter
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Install clippy
+        run: rustup component add clippy
+
+      - name: Run clippy
+        run: cargo clippy --all-targets --all-features -- -D warnings
+
+  format:
+    name: Run formatter
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v1
+      - run: rustup component add rustfmt
+      - run: cargo fmt --all -- --check
+
+  block-autosquash-commits:
```

Comment:
It doesn't matter so much but it's still nice to know if the tests passed regardless of this condition (for example, when you're still working on something)

Reply:
This job is executed separately from tests so it is possible to see if tests and build pass even if this fails.

Reply:
Oh, my bad, I didn't realize